### ISSUE TITLE
Translucent activity not finished with result

### ISSUE THREAD

**Issue by iandreyshev** [aartikov/Alligator#18]
Hi, I have `android:windowIsTranslucent` activity which should return result, but it happens only if rotate screen after call `FinishWithResult`.

### Example.1
**Open ForegroundActivity from BackgroundActivity**
unbind BackgroundActivity
bind ForegroundActivity
**Call FinishWithResult in ForegroundActivity**
unbind ForegroundActivity
bind BackgroundActivity
**Now we stay in the BackgroundActivity**

### Example.2
**Open ForegroundActivity from BackgroundActivity**
unbind BackgroundActivity
bind ForegroundActivity
**Rotate 1**
unbind ForegroundActivity
bind ForegroundActivity
bind BackgroundActivity
unbind BackgroundActivity
**_Call FinishWithResult, bat nothing happens. We stay at ForegroundActivity, but need BackgroundActivity_**
**Rotate 2**
unbind ForegroundActivity
bind ForegroundActivity
bind BackgroundActivity
unbind BackgroundActivity
unbind ForegroundActivity
bind BackgroundActivity
**Now we stay in the BackgroundActivity**

I think lifecycle for BackgroundActivity if we use `android:windowIsTranslucent` is not typical:
https://tech.pic-collage.com/something-about-android-windowistranslucent-641cc02ff66f

I have the same problem when test sample app from this repository. Is library knows about it ?

**Comment by aartikov** [comment]
Omg. Lifecycle of BackgroundActivity is really weird. Thank you for report.
Do you know any workarounds?

**Comment by iandreyshev** [comment]
@aartikov mb we need to save NavigationContexts in deque by push it in tail?

**Comment by aartikov** [comment]
@iandreyshev  May be, but I think it can be simpler. Like that:

```
public void bind(NavigationContext navigationContext) {
	if(mNavigationContext != null && mNavigationContext.getActivity() != navigationContext.getActivity()) {
		return;
	}
	...
}

public void unbind(Activity activity) {
	if(mNavigationContext != null && mNavigationContext.getActivity() != activity) {
		return;
	}
	...
}
```

These additional checks will prevent a navigation context to be changed if it is already bound.
It's a pity a little bit that `unbind` will require `Activity ` as an argument.

**Comment by aartikov** [comment]
Fixed in version 3.0.0
